File: spoj/MATRIZDIAGONAL/sol.py
# AC solution in C

import sys

order = int(input())
m = []
for i in range(order):
	m.append([int(x.strip()) for x in input().split()])

for i in range(order):
	for j in range(order):
		if i == j: continue
		if m[i][j] != 0:
			print('Falso')
			sys.exit(0)
			
print('Verdadero')

File: spoj/MATRIZDIAGONAL/sol.c
// AC solution in C

#include <stdio.h>

int inp() {
	int ch;
	do {
		ch = getchar_unlocked();
	}
	while(ch < '0' || ch > '9');

	int n=0;
	do{
		n *= 10;
		n += ch - '0';
		ch = getchar_unlocked();
	} while(ch >= '0' && ch <= '9');
	return n;
}

int main(){
  int m = inp();
  int falso = 0;
  for(int i=0; i<m; i++){
    for(int j=0; j<m; j++){
      int n = inp();
      if(i != j && n != 0){
		falso = 1;
        printf("Falso\n");
        break;
      }
    }
    if (falso) break;
  }
  if (!falso) printf("Verdadero\n");
  return 0;
}
